modules/table/src/main/java/org/apache/ignite/internal/table/TableImpl.java

```diff
@@ -413,4 +392,24 @@ private interface IndexStorageAdapterFactory {
         /** Creates the index decorator for given partition. */
         TableSchemaAwareIndexStorage create(int partitionId);
     }
+
+    /**
+     * Adds indexes to wait before inserting data into the table.
+     *
+     * @param indexIds Indexes Index IDs.
+     */
+    // TODO: IGNITE-19082 Needs to be redone/improved
```

Comment:
Did you discuss this with @AMashenkov (the author of this ticket) and that it will indeed be done under this ticket?

Reply:
Right now I'm close with the `IndexManager`, it should finish waiting for the indexes, since it will be removed in this ticket, then its author will need to take care of this problem.

Reply:
I'm just checking that the author is aware of the problem you are trying to fix and knows about this TODO